pulp_2to3_migration/app/pre_migration.py

```diff
@@ -355,8 +355,45 @@ def pre_migrate_all_without_content(plan):
             # filter by repo type
             repos_to_check = plan.type_to_repo_ids[plugin_plan.type]
 
-            mongo_repo_q = mongo_Q(repo_id__in=repos_to_check)
-            mongo_repo_qs = Repository.objects(mongo_repo_q)
+            # figure out which repositories/importers/distributors were updated since the last run
+            repo_premigrated_last = Pulp2Repository.objects.aggregate(
+                Max('pulp_last_updated')
+            )['pulp_last_updated__max'] or 0
+            imp_premigrated_last = Pulp2Importer.objects.aggregate(
+                Max('pulp_last_updated')
+            )['pulp_last_updated__max'] or 0
+            dist_premigrated_last = Pulp2Distributor.objects.aggregate(
+                Max('pulp_last_updated')
+            )['pulp_last_updated__max'] or 0
+
+            is_content_added_q = mongo_Q(last_unit_added__gte=repo_premigrated_last)
+            is_content_removed_q = mongo_Q(last_unit_removed__gte=repo_premigrated_last)
+            is_new_enough_repo_q = is_content_added_q | is_content_removed_q
+            is_empty_repo_q = (
+                mongo_Q(last_unit_added__exists=False) & mongo_Q(last_unit_removed__exists=False)
+            )
+            is_new_enough_imp_q = mongo_Q(last_updated__gte=imp_premigrated_last)
+            is_new_enough_dist_q = mongo_Q(last_updated__gte=dist_premigrated_last)
+            repo_id_q = mongo_Q(repo_id__in=repos_to_check)
+
+            updated_importers = Importer.objects(
+                repo_id_q & is_new_enough_imp_q
+            ).only('repo_id')
+            updated_imp_repos = set(imp.repo_id for imp in updated_importers)
+            updated_distributors = Distributor.objects(
+                repo_id_q & is_new_enough_dist_q
+            ).only('repo_id')
+            updated_dist_repos = set(dist.repo_id for dist in updated_distributors)
+            updated_impdist_repos = updated_imp_repos | updated_dist_repos
+            del updated_imp_repos
+            del updated_dist_repos
```

Comment:
?

Reply:
Just to get rid of extra copies of data  which is not needed anymore. I can keep it.

Reply:
It's not a problem.  If it's for memory purposes, it won't actually help.  If it's just to make it impossible to use the old copy by accident then it's worthwhile.